WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryPickerActivity.java

```diff
@@ -97,24 +96,29 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
         setContentView(R.layout.media_gallery_picker_layout);
-        mGridView = (GridView) findViewById(R.id.media_gallery_picker_gridview);
-        mGridView.setMultiChoiceModeListener(this);
-        mGridView.setOnItemClickListener(this);
+        mRecycler = (RecyclerView) findViewById(R.id.recycler);
+
+        int numColumns = MediaGridAdapter.getColumnCount(this);
+        mGridManager = new GridLayoutManager(this, numColumns);
+        mRecycler.setLayoutManager(mGridManager);
+
         // TODO: We want to inject the image loader in this class instead of using a static field.
-        mGridAdapter = new MediaGridAdapter(this, mSite, null, 0, WordPress.sImageLoader);
-        mGridAdapter.setSelectedItems(selectedItems);
+        mGridAdapter = new MediaGridAdapter(this, mSite, WordPress.sImageLoader);
         mGridAdapter.setCallback(this);
```

Comment:
Passing the activity as the callback to an adapter is fairly common practice, and I don't think there's a risk of a leak (but good you pointed it out so I could refresh my memory about it!).